common-rest/src/main/java/uk/ac/ebi/uniprot/api/rest/controller/BasicSearchController.java

```diff
@@ -49,13 +53,22 @@ protected BasicSearchController(ApplicationEventPublisher eventPublisher,
         MessageConverterContext<T> context = converterContextFactory.get(resource, contentType);
         context.setFields(fields);
         context.setEntityOnly(true);
-
         if (contentType.equals(LIST_MEDIA_TYPE)) {
             context.setEntityIds(Stream.of(getEntityId(entity)));
         } else {
             context.setEntities(Stream.of(entity));
         }
-        return ResponseEntity.ok()
+
+        Optional<String> redirectId = getEntityRedirectId(entity);
```

Comment:
I think lines 62 - 70 can be replaced by:
```java
    ResponseEntity.BodyBuilder responseBuilder = getEntityRedirectId(entity)
                .map(id -> ResponseEntity.status(HttpStatus.SEE_OTHER)
                        .header(HttpHeaders.LOCATION, getLocationURLForId(id)))
                .orElse(ResponseEntity.ok());
```

Reply:
I new you would comment on this.. LOL, but I never remember how to use it properly... tks!!!